### PR TITLE
Switch from ecdsa identity key and master key to sr25519

### PR DESCRIPTION
**Pull request by shelvenzhou**

### CODE REVIEW

**Comment by shelvenzhou** [comment]
This version passed the e2e tests of pRuntime and Gatekeeper.

File: standalone/pruntime/enclave/src/system/gk.rs [hunk 1]
```diff
@@ -607,15 +607,15 @@ where
             if let Some(master_key) = &self.state.master_key {
                 info!("Gatekeeper: try dispatch master key");
                 let derived_key = master_key
-                    .derive_secp256k1_pair(&[b"master_key"])
+                    .derive_sr25519_pair(&[b"master_key"])
```
Comment (h4x3rotab) [inline standalone/pruntime/enclave/src/system/gk.rs:610]: Why the derived key is by keyword "master_key"?

**Comment by h4x3rotab** [comment]
Please also check the failed test: <https://github.com/Phala-Network/phala-blockchain/pull/308/checks?check_run_id=3132171326>

File: standalone/pruntime/enclave/src/system/gk.rs [hunk 2]
```diff
@@ -593,12 +598,7 @@ where
                 info!("Gatekeeper: generate master key as the first gatekeeper");
                 // generate master key as the first gatekeeper
                 // no need to restart
-                self.state.set_master_key(
-                    crate::new_ecdsa_key().expect(
-                        "key generation should never fail since we give seed of correct sieze",
-                    ),
-                    false,
-                );
+                self.state.set_master_key(crate::new_sr25519_key(), false);
```
[the remark this answers is not in the document]
Comment (shelvenzhou) [reply standalone/pruntime/enclave/src/system/gk.rs:601]: is `rand::thread_rng` a cryptography RNG?

File: standalone/pruntime/enclave/src/system/gk.rs (hunk 1 above)
Comment (shelvenzhou) [reply standalone/pruntime/enclave/src/system/gk.rs:610]: fixed

File: standalone/pruntime/enclave/src/lib.rs [hunk 3]
```diff
@@ -1363,7 +1367,7 @@ fn get_info(_input: &Map<String, Value>) -> Result<Value, Value> {
     let pubkey = local_state
         .identity_key
         .as_ref()
-        .map(|pair| hex::encode(pair.public().as_ref()));
+        .map(|pair| hex::encode(pair.public().0.as_ref()));
```
[the remark this answers is not in the document]
Comment (shelvenzhou) [reply standalone/pruntime/enclave/src/lib.rs:1370]: fixed

**Review by h4x3rotab — APPROVED** [review]

**Comment by h4x3rotab** [comment]
LGTM

**Pull request merged**